runtime/src/test_utils.rs

```diff
@@ -0,0 +1,1279 @@
+
+        assert_eq!(exp.post, *post, "Unexpected PoSt verification");
+        if exp.exit_code != ExitCode::OK {
+            return Err(anyhow!(ActorError::unchecked(
+                exp.exit_code,
+                "Expected Failure".to_string(),
+            )));
+        }
+        Ok(())
+    }
+
+    fn verify_consensus_fault(
+        &self,
+        h1: &[u8],
+        h2: &[u8],
+        extra: &[u8],
+    ) -> anyhow::Result<Option<ConsensusFault>> {
+        let exp = self
+            .expectations
+            .borrow_mut()
+            .expect_verify_consensus_fault
+            .take()
+            .expect("Unexpected syscall to verify_consensus_fault");
+
+        if exp.require_correct_input {
+            assert_eq!(exp.block_header_1, h1, "Header 1 mismatch");
+            assert_eq!(exp.block_header_2, h2, "Header 2 mismatch");
+            assert_eq!(exp.block_header_extra, extra, "Header extra mismatch");
+        }
+        if exp.exit_code != ExitCode::OK {
+            return Err(anyhow!(ActorError::unchecked(
+                exp.exit_code,
+                "Expected Failure".to_string(),
+            )));
+        }
+        Ok(exp.fault)
+    }
+
+    fn batch_verify_seals(&self, batch: &[SealVerifyInfo]) -> anyhow::Result<Vec<bool>> {
+        let exp = self
+            .expectations
+            .borrow_mut()
+            .expect_batch_verify_seals
+            .take()
+            .expect("unexpected call to batch verify seals");
+        assert_eq!(exp.input.len(), batch.len(), "length mismatch");
+
+        for (i, exp_svi) in exp.input.iter().enumerate() {
+            assert_eq!(
+                exp_svi.sealed_cid, batch[i].sealed_cid,
+                "sealed CID mismatch at index {}",
+                i
+            );
+            assert_eq!(
+                exp_svi.unsealed_cid, batch[i].unsealed_cid,
+                "unsealed CID mismatch at index {}",
+                i
+            );
+        }
+        exp.result
+    }
+
+    fn verify_aggregate_seals(
+        &self,
+        aggregate: &AggregateSealVerifyProofAndInfos,
+    ) -> anyhow::Result<()> {
+        let exp = self
+            .expectations
+            .borrow_mut()
+            .expect_aggregate_verify_seals
+            .take()
+            .expect("unexpected call to verify aggregate seals");
+        assert_eq!(exp.in_svis.len(), aggregate.infos.len(), "length mismatch");
+        for (i, exp_svi) in exp.in_svis.iter().enumerate() {
+            assert_eq!(exp_svi.sealed_cid, aggregate.infos[i].sealed_cid, "mismatched sealed CID");
+            assert_eq!(
+                exp_svi.unsealed_cid, aggregate.infos[i].unsealed_cid,
+                "mismatched unsealed CID"
+            );
+        }
+        assert_eq!(exp.in_proof, aggregate.proof, "proof mismatch");
+        exp.result
+    }
+
+    fn verify_replica_update(&self, replica: &ReplicaUpdateInfo) -> anyhow::Result<()> {
```

Comment:
Verification related test utils can probably go.

Reply:
This one too, will break quite some code. I'll remove them later